src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -33,14 +33,15 @@ class RedirectController extends ContainerAware
      * In case the route name is empty, the status code will be 404 when permanent is false
      * and 410 otherwise.
      *
-     * @param Request       $request          The request instance
-     * @param string        $route            The route name to redirect to
-     * @param Boolean       $permanent        Whether the redirection is permanent
-     * @param Boolean|array $ignoreAttributes Whether to ignore attributes or an array of attributes to ignore
+     * @param Request       $request           The request instance
+     * @param string        $route             The route name to redirect to
+     * @param Boolean       $permanent         Whether the redirection is permanent
+     * @param Boolean|array $ignoreAttributes  Whether to ignore attributes or an array of attributes to ignore
+     * @param Boolean|array $ignoreQueryParams Whether to ignore query parameters or an array of query parameters to ignore
      *
      * @return Response A Response instance
      */
-    public function redirectAction(Request $request, $route, $permanent = false, $ignoreAttributes = false)
+    public function redirectAction(Request $request, $route, $permanent = false, $ignoreAttributes = false, $ignoreQueryParams = false)
```

Comment:
I would not make this configurable. For me, this is just a bug fix that should be fixed in 2.3.